lib/rspec/rails/matchers/active_job.rb

```diff
@@ -0,0 +1,72 @@
+require "active_job/base"
+
+module RSpec
+  module Rails
+    module Matchers
+      # Namespace for various implementations of ActiveJob features
+      #
+      # @api private
+      module ActiveJob
+        # @private
+        class HaveEnqueuedJobs < RSpec::Matchers::BuiltIn::BaseMatcher
+          def initialize(number, only = nil)
+            @number = number
+            @only   = only
+          end
+
+          def matches?(proc)
+            raise ArgumentError, "Proc have to be passed to expectation" unless Proc === proc
```

Comment:
A better wording for this would be "have_enqueued_jobs only supports block expectations"